pa_ringbuffer.c uses OSMemoryBarrier deprecated in macOS 10.12

```
 be@gsuss-iMac portaudio % cmake -D CMAKE_BUILD_TYPE=Debug -S . -B cmake_build

-- The C compiler identification is AppleClang 12.0.5.12050022
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found Regex: POSIX regex.h  
-- Found JACK: /usr/local/lib/libjack.dylib;Regex::regex  
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/be/sw/portaudio/cmake_build
be@gsuss-iMac portaudio % cmake --build cmake_build

[  5%] Building C object CMakeFiles/PortAudio.dir/src/common/pa_allocation.c.o
[ 11%] Building C object CMakeFiles/PortAudio.dir/src/common/pa_converters.c.o
[ 17%] Building C object CMakeFiles/PortAudio.dir/src/common/pa_cpuload.c.o
[ 23%] Building C object CMakeFiles/PortAudio.dir/src/common/pa_debugprint.c.o
[ 29%] Building C object CMakeFiles/PortAudio.dir/src/common/pa_dither.c.o
[ 35%] Building C object CMakeFiles/PortAudio.dir/src/common/pa_front.c.o
[ 41%] Building C object CMakeFiles/PortAudio.dir/src/common/pa_process.c.o
[ 47%] Building C object CMakeFiles/PortAudio.dir/src/common/pa_ringbuffer.c.o
/Users/be/sw/portaudio/src/common/pa_ringbuffer.c:131:9: warning: 'OSMemoryBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_thread_fence() from <stdatomic.h> instead [-Wdeprecated-declarations]
        PaUtil_FullMemoryBarrier(); /* (write-after-read) => full barrier */
        ^
/Users/be/sw/portaudio/src/common/pa_memorybarrier.h:68:40: note: expanded from macro 'PaUtil_FullMemoryBarrier'
#   define PaUtil_FullMemoryBarrier()  OSMemoryBarrier()
                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/usr/include/libkern/OSAtomicDeprecated.h:750:9: note: 'OSMemoryBarrier' has been explicitly marked deprecated here
void    OSMemoryBarrier( void );
        ^
/Users/be/sw/portaudio/src/common/pa_ringbuffer.c:144:5: warning: 'OSMemoryBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_thread_fence() from <stdatomic.h> instead [-Wdeprecated-declarations]
    PaUtil_WriteMemoryBarrier();
    ^
/Users/be/sw/portaudio/src/common/pa_memorybarrier.h:70:40: note: expanded from macro 'PaUtil_WriteMemoryBarrier'
#   define PaUtil_WriteMemoryBarrier() OSMemoryBarrier()
                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/usr/include/libkern/OSAtomicDeprecated.h:750:9: note: 'OSMemoryBarrier' has been explicitly marked deprecated here
void    OSMemoryBarrier( void );
        ^
/Users/be/sw/portaudio/src/common/pa_ringbuffer.c:181:9: warning: 'OSMemoryBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_thread_fence() from <stdatomic.h> instead [-Wdeprecated-declarations]
        PaUtil_ReadMemoryBarrier(); /* (read-after-read) => read barrier */
        ^
/Users/be/sw/portaudio/src/common/pa_memorybarrier.h:69:40: note: expanded from macro 'PaUtil_ReadMemoryBarrier'
#   define PaUtil_ReadMemoryBarrier()  OSMemoryBarrier()
                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/usr/include/libkern/OSAtomicDeprecated.h:750:9: note: 'OSMemoryBarrier' has been explicitly marked deprecated here
void    OSMemoryBarrier( void );
        ^
/Users/be/sw/portaudio/src/common/pa_ringbuffer.c:192:5: warning: 'OSMemoryBarrier' is deprecated: first deprecated in macOS 10.12 - Use atomic_thread_fence() from <stdatomic.h> instead [-Wdeprecated-declarations]
    PaUtil_FullMemoryBarrier();
    ^
/Users/be/sw/portaudio/src/common/pa_memorybarrier.h:68:40: note: expanded from macro 'PaUtil_FullMemoryBarrier'
#   define PaUtil_FullMemoryBarrier()  OSMemoryBarrier()
                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/usr/include/libkern/OSAtomicDeprecated.h:750:9: note: 'OSMemoryBarrier' has been explicitly marked deprecated here
void    OSMemoryBarrier( void );
        ^
4 warnings generated.
```

**Desktop (please complete the following information):**
 - OS: macOS 10.
 - OS Version: 11.2.3
 - PortAudio version: master branch 55b97bf16a80aaedc8db816ea6efb9e3bf88a803

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -286,7 +286,7 @@ elseif(UNIX)
     set(PKGCONFIG_CFLAGS "${PKGCONFIG_CFLAGS} -DPA_USE_COREAUDIO=1")
 
     # Use C11 so that we can make use of atomic library and avoid deprecation errors.
-    target_compile_definitions(PortAudio PRIVATE std=c11)
+    set_property(TARGET PortAudio PROPERTY C_STANDARD 11)
 
     set(PKGCONFIG_LDFLAGS_PRIVATE
       "${PKGCONFIG_LDFLAGS_PRIVATE} -framework CoreAudio -framework AudioToolbox -framework AudioUnit -framework CoreFoundation -framework CoreServices")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -284,6 +284,14 @@ elseif(UNIX)
     )
     target_compile_definitions(PortAudio PUBLIC PA_USE_COREAUDIO=1)
     set(PKGCONFIG_CFLAGS "${PKGCONFIG_CFLAGS} -DPA_USE_COREAUDIO=1")
+
+    include(CheckIncludeFiles)
+    check_include_files(stdatomic.h HAVE_STDATOMIC_H)
+    if(NOT HAVE_STDATOMIC_H)
+        target_compile_definitions(PortAudio PRIVATE PA_MAC_USE_OSS_MEMORY_BARRIER=1)
+        set(PKGCONFIG_CFLAGS "${PKGCONFIG_CFLAGS} -DPA_MAC_USE_OSS_MEMORY_BARRIER=1")
+    endif()
+
     set(PKGCONFIG_LDFLAGS_PRIVATE
       "${PKGCONFIG_LDFLAGS_PRIVATE} -framework CoreAudio -framework AudioToolbox -framework AudioUnit -framework CoreFoundation -framework CoreServices")
   else()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -285,11 +285,8 @@ elseif(UNIX)
     target_compile_definitions(PortAudio PUBLIC PA_USE_COREAUDIO=1)
     set(PKGCONFIG_CFLAGS "${PKGCONFIG_CFLAGS} -DPA_USE_COREAUDIO=1")
 
-    include(CheckIncludeFiles)
-    check_include_files(stdatomic.h HAVE_STDATOMIC_H)
-    if(NOT HAVE_STDATOMIC_H)
-        target_compile_definitions(PortAudio PRIVATE PA_MAC_USE_OS_MEMORY_BARRIER=1)
-    endif()
+    # Use C11 so that we can make use of atomic library and avoid deprecation errors.
+    target_compile_definitions(PortAudio PRIVATE std=c11)
 
     set(PKGCONFIG_LDFLAGS_PRIVATE
       "${PKGCONFIG_LDFLAGS_PRIVATE} -framework CoreAudio -framework AudioToolbox -framework AudioUnit -framework CoreFoundation -framework CoreServices")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -288,8 +288,7 @@ elseif(UNIX)
     include(CheckIncludeFiles)
     check_include_files(stdatomic.h HAVE_STDATOMIC_H)
     if(NOT HAVE_STDATOMIC_H)
-        target_compile_definitions(PortAudio PRIVATE PA_MAC_USE_OSS_MEMORY_BARRIER=1)
-        set(PKGCONFIG_CFLAGS "${PKGCONFIG_CFLAGS} -DPA_MAC_USE_OSS_MEMORY_BARRIER=1")
+        target_compile_definitions(PortAudio PRIVATE PA_MAC_USE_OS_MEMORY_BARRIER=1)
     endif()
 
     set(PKGCONFIG_LDFLAGS_PRIVATE

--- a/src/common/pa_memorybarrier.h
+++ b/src/common/pa_memorybarrier.h
@@ -61,13 +61,24 @@
  ****************/
 
 #if defined(__APPLE__)
-#   include <libkern/OSAtomic.h>
-    /* Here are the memory barrier functions. Mac OS X only provides
-       full memory barriers, so the three types of barriers are the same,
-       however, these barriers are superior to compiler-based ones. */
-#   define PaUtil_FullMemoryBarrier()  OSMemoryBarrier()
-#   define PaUtil_ReadMemoryBarrier()  OSMemoryBarrier()
-#   define PaUtil_WriteMemoryBarrier() OSMemoryBarrier()
+/* Support for stdatomic was added in XCode 7.
+ * If you need to build on an older version, define PA_MAC_USE_OS_MEMORY_BARRIER.
+ * TODO Find an automated way to check for stdatomic support. */
+#   if PA_MAC_USE_OS_MEMORY_BARRIER
+#       include <libkern/OSAtomic.h>
+        /* Here are the memory barrier functions. Mac OS X only provides
+           full memory barriers, so the three types of barriers are the same,
+           however, these barriers are superior to compiler-based ones.
+           These were deprecated in MacOS 10.12. */
+#       define PaUtil_FullMemoryBarrier()  OSMemoryBarrier()
+#       define PaUtil_ReadMemoryBarrier()  OSMemoryBarrier()
+#       define PaUtil_WriteMemoryBarrier() OSMemoryBarrier()
+#   else
+#       include <stdatomic.h>
+#       define PaUtil_FullMemoryBarrier()  atomic_thread_fence(memory_order_acq_rel)
+#       define PaUtil_ReadMemoryBarrier()  atomic_thread_fence(memory_order_acquire)
+#       define PaUtil_WriteMemoryBarrier() atomic_thread_fence(memory_order_release)
+#   endif
 #elif defined(__GNUC__)
     /* GCC >= 4.1 has built-in intrinsics. We'll use those */
 #   if (__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 1)

--- a/src/common/pa_memorybarrier.h
+++ b/src/common/pa_memorybarrier.h
@@ -61,10 +61,9 @@
  ****************/
 
 #if defined(__APPLE__)
-/* Support for stdatomic was added in XCode 7.
- * If you need to build on an older version, define PA_MAC_USE_OS_MEMORY_BARRIER.
+/* Support for the atomic library was added in C11.
  */
-#   if PA_MAC_USE_OS_MEMORY_BARRIER
+#   if (__STDC_VERSION__ < 201112L) || defined(__STDC_NO_ATOMICS__)
 #       include <libkern/OSAtomic.h>
         /* Here are the memory barrier functions. Mac OS X only provides
            full memory barriers, so the three types of barriers are the same,

--- a/src/common/pa_memorybarrier.h
+++ b/src/common/pa_memorybarrier.h
@@ -74,7 +74,7 @@
 #       define PaUtil_WriteMemoryBarrier() OSMemoryBarrier()
 #   else
 #       include <stdatomic.h>
-#       define PaUtil_FullMemoryBarrier()  atomic_thread_fence(memory_order_acq_rel)
+#       define PaUtil_FullMemoryBarrier()  atomic_thread_fence(memory_order_seq_cst)
 #       define PaUtil_ReadMemoryBarrier()  atomic_thread_fence(memory_order_acquire)
 #       define PaUtil_WriteMemoryBarrier() atomic_thread_fence(memory_order_release)
 #   endif

--- a/src/common/pa_memorybarrier.h
+++ b/src/common/pa_memorybarrier.h
@@ -63,7 +63,7 @@
 #if defined(__APPLE__)
 /* Support for stdatomic was added in XCode 7.
  * If you need to build on an older version, define PA_MAC_USE_OS_MEMORY_BARRIER.
- * TODO Find an automated way to check for stdatomic support. */
+ */
 #   if PA_MAC_USE_OS_MEMORY_BARRIER
 #       include <libkern/OSAtomic.h>
         /* Here are the memory barrier functions. Mac OS X only provides

